app/scripts/lib/id-management.js

```diff
@@ -24,7 +25,13 @@ function IdManagement (opts) {
   }
 
   this.signTx = function (txParams) {
+    //  calculate gas with custom gas multiplier
+    var gasMultiplier = txParams.gasMultiplier || 1
+    delete txParams.gasMultiplier
+    var gasPrice = parseFloat(new BN(ethUtil.stripHexPrefix(txParams.gasPrice), 16).toString()) * gasMultiplier
+    txParams.gasPrice = ethUtil.intToHex(parseInt(gasPrice))
```

Comment:
Here you're converting a hex to a BN to a string to a float, just to multiply by a number, and then convert back to hex.

Maybe it would be better to just convert to BN, and do all the work there.

Since the `gasMultiplier` is coming as a percentage, maybe treat a percent as a full number.

Note:  BN does not support decimals, so to multiply by a fraction, you'll need to bring it up and then take it down again.

An example of what I mean would be roughly (not checked):
```
var gasMultiplier = new BN((parseFloat(txParams.gasMultiplier) || 1) * 100, 16)
delete txParams.gasMultiplier
var gasPrice = new BN(txParams.gasPrice, 16).mul(gasMultiplier).div(new BN('100', 10))
txParams.gasPrice = gasPrice.toString(16)
```